sched/misc/nxsched_regs.c

```diff
@@ -0,0 +1,55 @@
+/****************************************************************************
+ * sched/misc/nxsched_regs.c
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <sys/stat.h>
+#include <sys/param.h>
+
+#include <nuttx/sched.h>
+#include "sched/sched.h"
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static uint8_t g_running_regs[XCPTCONTEXT_SIZE] aligned_data(16);
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+FAR uintptr_t *nxsched_regs(void)
```

Comment:
I'd prefer the caller provide the buffer for the register area, using a global might not be safe, unless you lock everything. I know the previous implementation did this and it is dangerous e.g. if multiple processes assert at the same time.